docs/manual/developer/04_styleguide.md

```diff
@@ -0,0 +1,100 @@
+# Styleguide
+
+## All files
+* Use UNIX style line endings
+* One blank line at the end of the file
+
+## Python
+* All Python files should follow [PEP 8](https://www.python.org/dev/peps/pep-0008/). We use
+[PEP 8 Speaks](https://pep8speaks.com/). We do make one change from PEP 8, our maximum
+line length is 99 characters. 
+* Methods should be defined before they called.
+
+## YAML
+* All YAML files shall use 4 spaces inplace of tabs.
```

Comment:
Interestingly, YAML forbids tabs and prefers 2 spaces: https://docs.saltproject.io/en/latest/topics/troubleshooting/yaml_idiosyncrasies.html#spaces-vs-tabs 

Thoughts? Though, admittedly there isn't a better source than this that I can find easily.

Reply:
We are inconstant across the codebase. We need to pick one and stick with it. I'm fine with two if that the way we want to go.

Reply:
Let's pick a style that is least disruptive - you don't have to have the history cluttered by reformatting when you are trying to find out what the historical reasons of why the code looks in some particular way.

Reply:
I added a note about new files being four and existing files can be two spaces.